xds/internal/resolver/serviceconfig.go

```diff
@@ -142,7 +142,7 @@ type erroringConfigSelector struct {
 }
 
 func newErroringConfigSelector(err error, xdsNodeID string) *erroringConfigSelector {
-	return &erroringConfigSelector{err: annotateErrorWithNodeID(status.Errorf(codes.Unavailable, err.Error()), xdsNodeID)}
+	return &erroringConfigSelector{err: annotateErrorWithNodeID(status.Errorf(codes.Unavailable, "%s", err.Error()), xdsNodeID)}
```

Comment:
```suggestion
	return &erroringConfigSelector{err: annotateErrorWithNodeID(status.Error(codes.Unavailable, err.Error()), xdsNodeID)}
```

My suggestion is style vs. your fix, but the fix is a bug, not style.  If `err.Error()` outputs `%s` or something, then it would print weird error-looking stuff instead.

https://go.dev/play/p/BxccGnfgqas

```
Hello, %!s(MISSING)
```

Reply:
I see. thanks! changed.